### PR TITLE
Improve Intel cmake compiler flags

### PR DESCRIPTION
**Pull request by minyez**

### CODE REVIEW

File: cmake/CompilerFlags.cmake [hunk 1]
```diff
@@ -23,10 +18,16 @@ set(GCC_DEBUG
 set(GCC_RELEASE -O3)  # Level 3 optimisation. Could also consider -fpack-derived
 
 # Intel 
+set(INTEL_BASE
+    -stand f08          # Fortran standard set to 2008
```
Comment (AlexBuccheri) [inline cmake/CompilerFlags.cmake:22]: Comment's wrong for this, `stand` warns w.r.t. use of non-standard fortran. See [here](https://www.intel.com/content/www/us/en/docs/fortran-compiler/developer-guide-reference/2023-2/stand.html)

This should not be a base flag

File: cmake/CompilerFlags.cmake [hunk 2]
```diff
@@ -23,10 +18,16 @@ set(GCC_DEBUG
...
+    -implicitnone       # Give warning when undeclared variable is used
```
Comment (AlexBuccheri) [inline cmake/CompilerFlags.cmake:23]: Perhaps a more appropriate description is:
"Sets the default type of a variable to undefined"

File: cmake/CompilerFlags.cmake [hunk 3]
```diff
@@ -23,10 +18,16 @@ set(GCC_DEBUG
...
+    -diag-disable=5268  # Remove warning: Extension to standard: The text exceeds right hand column allowed on the line.
+    -free               # Specify free-fortmat
+   )
 set(INTEL_DEBUG 
     -g           # Generate symbols
     -traceback   # symbolic stack traceback
-    -fp          # Disables the ebp register in optimizations and sets the ebp register to be used as the frame pointer.
+    # -fp          # Disables the ebp register in optimizations and sets the ebp register to be used as the frame pointer.
```
Comment (AlexBuccheri) [inline cmake/CompilerFlags.cmake:30]: If this is depreciated, it should be deleted, not commented-out.

File: cmake/CompilerFlags.cmake [hunk 4]
```diff
@@ -23,10 +18,16 @@ set(GCC_DEBUG
 set(GCC_RELEASE -O3)  # Level 3 optimisation. Could also consider -fpack-derived
 
 # Intel 
+set(INTEL_BASE
+    -stand f08          # Fortran standard set to 2008
+    -implicitnone       # Give warning when undeclared variable is used
+    -diag-disable=5268  # Remove warning: Extension to standard: The text exceeds right hand column allowed on the line.
+    -free               # Specify free-fortmat
+   )
```
Comment (AlexBuccheri) [inline cmake/CompilerFlags.cmake:26]: @minyez The general question is, which verison/s of ifort are these valid for? Ifort classic/LLVM looks like the flags are somewhat inconsistent with what's here.

@Panadestein  You fancy adding an Intel build to the CI matrix in this PR? Else nothing can be validated.

Comment (minyez) [reply cmake/CompilerFlags.cmake:26]: These options are valid for both ifort and ifx (LLVM), tested in oneAPI versions 2022 update 1 and 2023 update 1. On the other hand, some check options specified in the release flags are not implemented in the early version of ifx.

File: cmake/CompilerFlags.cmake (hunk 1 above)
Comment (minyez) [reply cmake/CompilerFlags.cmake:22]: Thanks for pointing out the correct meaning of `-stand`.

The reason I added it as an intel base flag is to correspond to the original `-std='f2008'` flag for GCC. The other flags in `INTEL_BASE` also follow this intention.

**Comment by AlexBuccheri** [comment]
@minyez I've added Intel one-API to the CI using this [reference](https://gist.github.com/scivision/d94bb10a01fa3b8ed0c9a93ee16318ba). This way, we can test flag validity

**Comment by minyez** [comment]
@AlexBuccheri Hi Alex, I modified the github action to enable test with both ifx and ifort and accelarate oneAPI by caching the oneAPI install (refering [this sample](https://github.com/oneapi-src/oneapi-ci/blob/master/.github/workflows/build_all.yml) ).

The build process looks good and both minimax tests pass. But the unittest of analytic continuation component failed with NaN output by `pade` rather than the correct 0.5-0.5i.

**Pull request merged**